### PR TITLE
[v16] feat: Disable auto-enroll via environment variable

### PR DESCRIPTION
**Pull request by codingllama**
Backport #47679 and #47723 to branch/v16

changelog: Auto-enroll may be locally disabled using the TELEPORT_DEVICE_AUTO_ENROLL_DISABLED=1 environment variable

### CODE REVIEW

**Review by ryanclark — APPROVED** [review]

**Comment by codingllama** [comment]
This is flaky! Mailed #47723, will merge it here once it lands.

- #47723

**Comment by codingllama** [comment]
Cherry-picked #47723 in here. I'll do the same for the other backports and queue them all.

**Comment by codingllama** [comment]
PTAL @rosstimothy.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**
